airflow/providers/cncf/kubernetes/sensors/spark_kubernetes.py

```diff
@@ -37,38 +39,75 @@ class SparkKubernetesSensor(BaseSensorOperator):
     :type namespace: str
     :param kubernetes_conn_id: the connection to Kubernetes cluster
     :type kubernetes_conn_id: str
+    :param attach_log: determines whether logs for driver pod should be appended to the sensor log
+    :type attach_log: bool
     """
 
-    template_fields = ('application_name', 'namespace')
-    FAILURE_STATES = ('FAILED', 'UNKNOWN')
-    SUCCESS_STATES = ('COMPLETED',)
+    template_fields = ("application_name", "namespace")
+    FAILURE_STATES = ("FAILED", "UNKNOWN")
+    SUCCESS_STATES = ("COMPLETED",)
 
     @apply_defaults
-    def __init__(self, *,
-                 application_name: str,
-                 namespace: Optional[str] = None,
-                 kubernetes_conn_id: str = 'kubernetes_default',
-                 **kwargs):
+    def __init__(
+        self,
+        application_name: str,
```

Comment:
```suggestion
        self,
        *,
        application_name: str,
```
See: https://github.com/apache/airflow/issues/9942